contracts/RouterStrategy.sol

```diff
@@ -0,0 +1,227 @@
+            _debtPayment = Math.min(_debtOutstanding, _amountFreed);
+            if (_loss > 0) {
+                _profit = 0;
+            }
+        }
+    }
+
+    function adjustPosition(uint256 _debtOutstanding) internal override {
+        if (emergencyExit) {
+            return;
+        }
+
+        uint256 balance = balanceOfWant();
+        if (balance > 0) {
+            _checkAllowance(address(yVault), address(want), balance);
+            yVault.deposit();
+        }
+    }
+
+    function liquidatePosition(uint256 _amountNeeded)
+        internal
+        override
+        returns (uint256 _liquidatedAmount, uint256 _loss)
+    {
+        uint256 balance = balanceOfWant();
+        if (balance >= _amountNeeded) {
+            return (_amountNeeded, 0);
+        }
+
+        uint256 toWithdraw = _amountNeeded.sub(balance);
+        _withdrawFromYVault(toWithdraw);
+
+        uint256 totalAssets = want.balanceOf(address(this));
+        if (_amountNeeded > totalAssets) {
+            _liquidatedAmount = totalAssets;
+            _loss = _amountNeeded.sub(totalAssets);
+        } else {
+            _liquidatedAmount = _amountNeeded;
+        }
+    }
+
+    function _withdrawFromYVault(uint256 _amount) internal {
+        if (_amount == 0) {
+            return;
+        }
+
+        uint256 sharesToWithdraw =
+            Math.min(
+                _investmentTokenToYShares(_amount),
+                yVault.balanceOf(address(this))
+            );
+        if (sharesToWithdraw == 0) {
+            return;
+        }
+
+        yVault.withdraw(sharesToWithdraw, address(this), maxLoss);
+    }
+
+    function liquidateAllPositions()
+        internal
+        override
+        returns (uint256 _amountFreed)
+    {
+        (_amountFreed, ) = liquidatePosition(
+            vault.strategies(address(this)).totalDebt
+        );
+    }
+
+    function prepareMigration(address _newStrategy) internal override {
+        IERC20(yVault).safeTransfer(
+            _newStrategy,
+            IERC20(yVault).balanceOf(address(this))
+        );
+    }
+
+    function protectedTokens()
+        internal
+        view
+        override
+        returns (address[] memory)
+    {}
+
+    function ethToWant(uint256 _amtInWei)
+        public
+        view
+        virtual
+        override
+        returns (uint256)
+    {
+        return _amtInWei;
+    }
+
+    function setMaxLoss(uint256 _maxLoss) public onlyEmergencyAuthorized {
+        maxLoss = _maxLoss;
+    }
+
+    function _takeVaultProfit() internal {
+        uint256 _debt = vault.strategies(address(this)).totalDebt;
+        uint256 _valueInVault = valueOfInvestment();
+        if (_debt >= _valueInVault) {
+            return;
+        }
+
+        uint256 _profit = _valueInVault.sub(_debt);
+        uint256 _ySharesToWithdraw = _investmentTokenToYShares(_profit);
+        if (_ySharesToWithdraw > 0) {
+            yVault.withdraw(_ySharesToWithdraw, address(this), maxLoss);
```

Comment:
Is there some mechanism to make sure there are no losses when withdrawn during the sandwich protection window? More just curious how it works here.

Reply:
Good point. 
No protection, only us calling it on the right time.

If we set deposits to 0, we can do it every week and it's pretty safe.